app/models/user.rb

```diff
@@ -82,6 +82,8 @@ class User < ApplicationRecord
     message: "can only contain letters, numbers, and underscores"
   }, allow_nil: true, length: { within: 0..20 }
 
+  validates_formatting_of :homepage_url, using: :url, allow_blank: true
```

Comment:
When I do `allow_nil: true` twitter tests in the profile_test.rb would fail. Seeking advice if this is best practice.

Reply:
ah it's because homepage_url is optional, but the form always submits an empty string. I forget if there's a good way to handle this

Reply:
I think this is expected.